int32 load_state
int32 unload_state